### PR TITLE
[Tower only] Make pendo license settings opt out whenever license is added

### PR DESCRIPTION
**Pull request by jlmitch5**
**Note that this will not affect AWX users in any way, as all the license-related Tower screens and process do not apply to AWX.  This (and all Pendo tracking) is disabled for AWX.**

This adds a checkbox so that optional Pendo tracking for Tower is checked by default.  This can be revoked at any time through resubmitting the license on the license screen.

Also some clean up to the license controllers as the styling was a bit malformed.

### CODE REVIEW

File: awx/ui/client/src/login/authenticationServices/pendo.service.js [hunk 1]
```diff
@@ -107,24 +98,44 @@ export default
...
+                                // reject function for setRole
+                                $log.debug(reason);
+                            });
+                } else {
                     $log.debug('Pendo is turned off.');
                 }
-             }
+            },
+
+            updatePendoTrackingState: function(tracking_type) {
```
Comment (jlmitch5) [inline awx/ui/client/src/login/authenticationServices/pendo.service.js:131]: this is the new function

File: awx/ui/client/src/license/license.controller.js [hunk 2]
```diff
@@ -10,106 +10,123 @@ export default
...
-		};
-     }
-];
+                    ConfigService.getConfig()
+                        .then(function(config) {
+                            delete($rootScope.features);
+                            FeaturesService.get();
+
+                            if ($scope.newLicense.pendo) {
+                                pendoService.updatePendoTrackingState('detailed');
```
Comment (jlmitch5) [inline awx/ui/client/src/license/license.controller.js:123]: this is where the settings PATCH happens

File: awx/ui/client/src/license/license.controller.js [hunk 3]
```diff
@@ -10,106 +10,123 @@ export default
...
             $scope.license = config;
             $scope.license.version = config.version.split('-')[0];
             $scope.time = {};
             $scope.time.remaining = calcDaysRemaining($scope.license.license_info.time_remaining);
             $scope.time.expiresOn = calcExpiresOn($scope.license.license_info.license_date);
             $scope.valid = CheckLicense.valid($scope.license.license_info);
             $scope.compliant = $scope.license.license_info.compliant;
+            $scope.newLicense = {};
+
+            $scope.newLicense.pendo = true;
```
Comment (jlmitch5) [inline awx/ui/client/src/license/license.controller.js]: this is how I'm setting the checkbox to be checked by default

**Review by jlmitch5 — COMMENTED** [review]
Non-reformatting additions marked with comments

File: awx/ui/client/src/license/license.partial.html [hunk 4]
```diff
@@ -122,6 +122,19 @@
...
 					</div>
+					<div class="form-group License-detailsGroup License-detailsGroup--withSeparator">
+						<div class="checkbox">
+							<label class="License-details--label">
+								<input type="checkbox" ng-model="newLicense.pendo" ng-disabled="!user_is_superuser" required>
+								<translate>By default, Tower collects and transmits analytics data on Tower usage to Red Hat. This data is used to enhance future releases of the Tower Software and help streamline customer experience and success. For more information, see
+										<a target="_blank"
+												href="http://docs.ansible.com/ansible-tower/latest/html/installandreference/user-data.html#index-0">
+												this Tower documentation page
+										</a>.
```
Comment (jaredevantabor) [inline awx/ui/client/src/license/license.partial.html]: There's an extra space b/t "page" and the period here at the end of the sentence. 

Also a very small nitpick, but in the card there's an extra sentence at the end: "Uncheck this box to disable this feature." Without this sentence it's not obvious what action the user is taking by checking/unchecking the box.

File: awx/ui/client/src/login/authenticationServices/pendo.service.js [hunk 5]
```diff
@@ -5,87 +5,84 @@
...
-                    // AccountId: hash of license_key from license
-
+                // config.deployment_id is a hash of the tower license_key
                 options.visitor.id = $rootScope.current_user.id + '@' + config.deployment_id;
                 options.account.id = config.deployment_id;
             },
 
+            // Anonymous mode sends:
+            // VisitorId: <hardcoded id that is the same across all anonymous>
+            // AccountId: <hardcoded id that is the same across all anonymous>
```
Comment (jaredevantabor) [inline awx/ui/client/src/login/authenticationServices/pendo.service.js:52]: Thanks for adding these back in!

File: awx/ui/client/src/license/license.controller.js [hunk 6]
```diff
@@ -10,106 +10,125 @@ export default
...
-        var init = function(config){
+        const init = function(config) {
             // license/license.partial.html compares fileName
             $scope.fileName = N_("No file selected.");
-            $scope.title = $rootScope.licenseMissing ? ($rootScope.BRAND_NAME + i18n._(" License")) : i18n._("License Management");
+
+            if ($rootScope.licenseMissing) {
+                $scope.title = $rootScope.BRAND_NAME + i18n._(" License");
+            } else {
+                $scope.title = i18n._("License Management")
```
Comment (jaredevantabor) [inline awx/ui/client/src/license/license.controller.js]: missing semicolon 

client/src/license/license.controller.js
  line 46  col 60  Missing semicolon.

  ⚠  1 warning

**Review by jaredevantabor — APPROVED** [review]
Thanks for doing this, and putting up w/ me in our discussions about it. I think it'd be fine to do https://github.com/ansible/ansible-tower/issues/5866 in a follow up PR.

File: awx/ui/client/src/license/license.partial.html [hunk 7]
```diff
@@ -130,7 +130,7 @@
 										<a target="_blank"
 												href="http://docs.ansible.com/ansible-tower/latest/html/installandreference/user-data.html#index-0">
 												this Tower documentation page
-										</a>.
+										</a>.  Uncheck this box to disable this feature.
```
Comment (jaredevantabor) [inline awx/ui/client/src/license/license.partial.html]: thank you thank you

**Review by jaredevantabor — APPROVED** [review]

**Pull request merged**